llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -4,13 +4,22 @@
 
 from llama_index.core.workflow.decorators import step, StepConfig
 from llama_index.core.workflow.events import StartEvent, StopEvent, Event
-from llama_index.core.workflow.utils import get_steps_from_class
+from llama_index.core.workflow.utils import (
+    get_steps_from_class,
+    get_steps_from_instance,
+)
 
 
 from .errors import WorkflowRuntimeError, WorkflowTimeoutError, WorkflowValidationError
 
 
-class Workflow:
+class _WorkflowMeta(type):
```

Comment:
This is needed so that each subclass will have its own isolated storage of `_step_functions`. Otherwise, being dictionary mutable the storage would be shared across any workflow type.

Reply:
Did I miss it or is this used in the context PR?

Reply:
It's used as the metaclass for the `Workflow` base class https://github.com/masci/llama_index/pull/3/files#diff-a6645f3a6c73a691c2f44c9235a6217c65e099747acb1d9f9fc1b7cf5ec158c0R22

Reply:
Ohhh there it is, nice